AlfrescoApp/Views/Action Collection View/Model/ActionCollectionItem.m

```diff
@@ -172,6 +174,90 @@ - (void)dealloc
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierEmailAsLink])
+    {
+        returnIdentifier = kActionCollectionItemEmailAsLinkButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierOpenIn])
+    {
+        returnIdentifier = kActionCollectionItemOpenInButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierLike])
+    {
+        returnIdentifier = kActionCollectionItemLikeButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierUnlike])
+    {
+        returnIdentifier = kActionCollectionItemUnlikeButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierFavourite])
+    {
+        returnIdentifier = kActionCollectionItemFavouriteButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierUnfavourite])
+    {
+        returnIdentifier = kActionCollectionItemUnfavouriteButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierComment])
+    {
+        returnIdentifier = kActionCollectionItemCommentButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierEdit])
+    {
+        returnIdentifier = kActionCollectionItemEditButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierDownload])
+    {
+        returnIdentifier = kActionCollectionItemDownloadButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierPrint])
+    {
+        returnIdentifier = kActionCollectionItemPrintButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierDelete])
+    {
+        returnIdentifier = kActionCollectionItemDeleteButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierRename])
+    {
+        returnIdentifier = kActionCollectionItemRenameButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierUploadDocument])
+    {
+        returnIdentifier = kActionCollectionItemUploadButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierSendForReview])
+    {
+        returnIdentifier = kActionCollectionItemSendForReviewButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierCreateSubfolder])
+    {
+        returnIdentifier = kActionCollectionItemSubFolderButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierUploadNewVersion])
+    {
+        returnIdentifier = kActionCollectionItemUploadNewVersionButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierSync])
+    {
+        returnIdentifier = kActionCollectionItemSyncButton;
+    }
+    else if ([itemIdentifier isEqualToString:kActionCollectionIdentifierUnsync])
+    {
+        returnIdentifier = kActionCollectionItemUnsyncButton;
+    }
+    
+    return returnIdentifier;
+}
+
```

Comment:
It would be nice and also a performance improvement to avoid huge nested ifs like this one. A dictionary would do the job.